### PR TITLE
[OpenVINO] Update to 2024.1.0

### PR DESCRIPTION
**Pull request by culhatsker**

### CODE REVIEW

File: ports/openvino/portfile.cmake [hunk 1]
```diff
@@ -2,7 +2,7 @@ vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO openvinotoolkit/openvino
     REF "${VERSION}"
-    SHA512 89c60aa8c73d91003f9fdd4e8f93213b98164ecdcd4a4264ba93352473666a290d1dd55b61cf9c5d62b0cef429f989bb31f5d4bfa67cf6d97dfc2abca45d4186
+    SHA512 de6392da7722c7da2b626b9d2d7c2d5ace91bed8a3c9a6fe164c29fb7180ba19a43055f76bfa588f24de8641d9f90a0dc06ff0c80ebd7be1a5b7b62db8992781
```
Comment (artanokhov) [inline ports/openvino/portfile.cmake:5]: Correct hash
c1e1c119ef03c38e45dddc4db74a656fd8c4f922d8106ca0b0251ad3f33de1b7bf48777dc58b9ad05518eaccb91d302e66e9a49844b396b347defdee6438d176

File: ports/openvino/portfile.cmake [hunk 2]
```diff
@@ -139,3 +156,7 @@ vcpkg_install_copyright(
         "OpenVINO License")
 
 file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
+
+if(VCPKG_TARGET_IS_WINDOWS AND VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin/ze_loader.dll" "${CURRENT_PACKAGES_DIR}/debug/bin/ze_loader.dll")
+endif()
```
Comment (culhatsker) [inline ports/openvino/portfile.cmake:162]: This file is only used during build and shouldn't be in the install directory, but for some reason it is.
I wonder if it's acceptable way of working this around.

Comment (culhatsker) [reply ports/openvino/portfile.cmake]: Disregard, this was a mistake, it is in fact needed, but can't be statically linked, removing support for NPU when statically linked

File: ports/openvino/vcpkg.json [hunk 3]
```diff
@@ -56,6 +56,10 @@
     },
     "hetero",
     "ir",
+    {
+      "name": "npu",
+      "platform": "x64 & !(osx | uwp)"
```
Comment (ilya-lavrenov) [inline ports/openvino/vcpkg.json:61]:
```suggestion
      "platform": "x64 & !(osx | uwp) & !static"
```

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by jimwang118** [comment]
If you confirm the repair is complete, click "ready for review".

File: ports/openvino/vcpkg.json [hunk 4]
```diff
@@ -56,6 +56,10 @@
     },
     "hetero",
     "ir",
+    {
+      "name": "npu",
+      "platform": "x64 & !(osx | uwp) & !static"
```
Comment (ilya-lavrenov) [inline ports/openvino/vcpkg.json:61]:
```suggestion
      "platform": "x64 & !(osx | uwp | static)"
```

**Comment by jimwang118** [comment]
All features passed with following triplets:

```
x86-windows
x64-windows
x64-windows-static
```

**Comment by culhatsker** [comment]
@jimwang118, is there anything else we need to do to merge these changes?

**Review by artanokhov — APPROVED** [review]

File: ports/openvino/vcpkg.json [hunk 5]
```diff
@@ -56,6 +56,10 @@
...
+    },
```
Comment (JavierMatosD) [inline ports/openvino/vcpkg.json:62]: Does this really need to be a `default-feature`? We typically only add stuff to the `default-features` if it is essentially required. See - https://learn.microsoft.com/en-us/vcpkg/concepts/default-features#role-of-default-features

Comment (culhatsker) [reply ports/openvino/vcpkg.json]: Well, you're right, it is not essential, removed it from defaults

**Comment by culhatsker** [comment]
x64_osx build failed with a few instances of this error:
> pwsh(93825) MallocStackLogging: can't turn off malloc stack logging because it was not enabled.

But it seems to be an issue related to powershell, not this PR. Please correct me if I'm wrong.

**Review by data-queue — APPROVED** [review]

**Comment by culhatsker** [comment]
@JavierMatosD , may I ask you to have another look?

**Pull request merged**